sdk/eventhub/Azure.Messaging.EventHubs/src/EventProcessorClient.cs

```diff
@@ -522,146 +882,339 @@ protected override async ValueTask InitializeProcessingForPartitionAsync(Partiti
+                // on the next load balancing loop as long as this instance still owns the partition.
+                // TODO: delegate the exception handling to an Exception Callback.  Do we really need a try-catch here?
+            }
         }
 
         /// <summary>
-        ///   Performs the task needed to clean up resources used by the <see cref="EventProcessorClient" />,
-        ///   including ensuring that the event client itself has been closed.
+        ///   Stops an owned partition processing task in case it is running.  It is also removed from the tasks dictionary
+        ///   along with its corresponding token source.
         /// </summary>
         ///
+        /// <param name="partitionId">The identifier of the Event Hub partition whose processing is being stopped.</param>
+        /// <param name="reason">The reason why the processing for the specified partition is being stopped.</param>
+        ///
         /// <returns>A task to be resolved on when the operation has completed.</returns>
         ///
-        public virtual async ValueTask DisposeAsync() => await CloseAsync().ConfigureAwait(false);
+        private async Task StopPartitionProcessingIfRunningAsync(string partitionId,
+                                                                 ProcessingStoppedReason reason)
+        {
+            if (ActivePartitionProcessors.TryRemove(partitionId, out var processingTask)
+                && ActivePartitionProcessorTokenSources.TryRemove(partitionId, out var tokenSource))
+            {
+                try
+                {
+                    tokenSource.Cancel();
+                    await processingTask.ConfigureAwait(false);
+                }
+                catch (Exception ex) when (ex is TaskCanceledException || ex is OperationCanceledException)
+                {
+                    // Nothing to do here.  These exceptions are expected.
+                }
+                catch (Exception)
+                {
+                    // TODO: delegate the exception handling to an Exception Callback.
+                }
+                finally
+                {
+                    tokenSource.Dispose();
+                }
+            }
 
-        /// <summary>
-        ///   Determines whether the specified <see cref="System.Object" /> is equal to this instance.
-        /// </summary>
-        ///
-        /// <param name="obj">The <see cref="System.Object" /> to compare with this instance.</param>
-        ///
-        /// <returns><c>true</c> if the specified <see cref="System.Object" /> is equal to this instance; otherwise, <c>false</c>.</returns>
-        ///
-        [EditorBrowsable(EditorBrowsableState.Never)]
-        public override bool Equals(object obj) => base.Equals(obj);
+            // TODO: if reason = Shutdown or OwnershipLost and we got an exception when closing, what should the final reason be?
+
+            PartitionContexts.TryRemove(partitionId, out var context);
```

Comment:
if we're not using `context`, we may want to make that `_` to indicate that it's a throw-away.

Reply:
We are using it.

https://github.com/Azure/azure-sdk-for-net/blob/68b1a07746e77556f7c5bf64f5f387acce37b747/sdk/eventhub/Azure.Messaging.EventHubs/src/EventProcessorClient.cs#L1073-L1078